src/main/cpp/t07_ip.cpp

```diff
@@ -29,10 +29,31 @@
 
 #include "t07_ip.h"
 #include <iostream>
-
+#include <string>
 
 using namespace std;
 
 int t07_ip() {
-
+	string str, temp;
+	char c;
+	int it = 0;
+	bool fl = false;
+	cin >> str;
+	for (size_t i = 0; i <= str.length(); i++) {
+		if (isdigit(str[i]))
+			temp.push_back(str[i]);
+		else if ((str[i] == '.' || str[i] == '\0') && !temp.empty() && it <= 4) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [When i==str.size(), str[i] is out of bounds.](https://app.codacy.com/app/thejerome/IFMO_CPP_programming_20190915/pullRequest?prid=4685590)